### PR TITLE
[MIG] account_invoice_transmit_method: Migration to 14.0

### PR DESCRIPTION
**Pull request by phuctranfxvn**
Requires:
- base_view_inheritance_extension migrated in https://github.com/OCA/server-tools/pull/1937

### CODE REVIEW

File: account_invoice_transmit_method/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        " ('supplier_ok', '=', True)]",
+    )
+    # Field used to match specific invoice transmit method
+    # to show/display fields/buttons, add constraints, etc...
+    transmit_method_code = fields.Char(related="transmit_method_id.code", store=True)
+    transmit_method_domain_sale = fields.Boolean(
+        compute="_compute_transmit_method_domain", default=True
+    )
+    transmit_method_domain_purchase = fields.Boolean(
+        compute="_compute_transmit_method_domain", default=True
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py]: What is the goal of default value on a computed field not stored?
For me it's useless, but maybe I'm wrong?

File: account_invoice_transmit_method/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            else:
+                record.transmit_method_domain_sale = True
+                record.transmit_method_domain_purchase = True
+
+    @api.onchange("partner_id", "company_id")
+    def _transmit_method_partner_change(self):
+        if self.partner_id and self.move_type:
+            if self.move_type in ("out_invoice", "out_refund"):
+                self.transmit_method_id = (
+                    self.partner_id.customer_invoice_transmit_method_id.id or False
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py]: No need to set the `.id`, set the record set works too, and no need `or False`: it's already done by the ORM.
```suggestion
                    self.partner_id.customer_invoice_transmit_method_id
```

File: account_invoice_transmit_method/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                )
+            else:
+                self.transmit_method_id = (
+                    self.partner_id.supplier_invoice_transmit_method_id.id or False
```
Comment (jcoux) [inline account_invoice_transmit_method/models/account_move.py]: No need to set the `.id`, set the record set works too, and no need `or False`: it's already done by the ORM.
```suggestion
                    self.partner_id.supplier_invoice_transmit_method_id
```

File: account_invoice_transmit_method/models/partner.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        "transmit.method",
+        string="Customer Invoice Transmission Method",
+        company_dependant=True,
+        tracking=True,
+        ondelete="restrict",
+    )
+    customer_invoice_transmit_method_code = fields.Char(
+        related="customer_invoice_transmit_method_id.code",
+        string="Customer Invoice Transmission Method Code",
+        readonly=True,
```
Comment (jcoux) [inline account_invoice_transmit_method/models/partner.py]: related fields are readonly by default.
```suggestion
```

File: account_invoice_transmit_method/models/partner.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        "transmit.method",
+        string="Vendor Invoice Reception Method",
+        company_dependant=True,
+        tracking=True,
+        ondelete="restrict",
+    )
+    supplier_invoice_transmit_method_code = fields.Char(
+        related="supplier_invoice_transmit_method_id.code",
+        string="Vendor Invoice Reception Method Code",
+        readonly=True,
```
Comment (jcoux) [inline account_invoice_transmit_method/models/partner.py]: related fields are readonly by default.
```suggestion
```

**Review by jcoux — COMMENTED** [review]
Technical review only.

File: account_invoice_transmit_method/models/account_move.py (hunk 1 above)
Comment (phuctranfxvn) [reply account_invoice_transmit_method/models/account_move.py]: Yes you are right. actually this exists in the previous version. removed in https://github.com/OCA/account-invoicing/pull/861/commits/da1b589510f39991f7a9c89da593346766b47c79

File: account_invoice_transmit_method/models/account_move.py (hunk 2 above)
Comment (phuctranfxvn) [reply account_invoice_transmit_method/models/account_move.py]: fixed in https://github.com/OCA/account-invoicing/pull/861/commits/da1b589510f39991f7a9c89da593346766b47c79

File: account_invoice_transmit_method/models/account_move.py (hunk 3 above)
Comment (phuctranfxvn) [reply account_invoice_transmit_method/models/account_move.py]: fixed in https://github.com/OCA/account-invoicing/pull/861/commits/da1b589510f39991f7a9c89da593346766b47c79

File: account_invoice_transmit_method/models/partner.py (hunk 4 above)
Comment (phuctranfxvn) [reply account_invoice_transmit_method/models/partner.py]: fixed in https://github.com/OCA/account-invoicing/pull/861/commits/da1b589510f39991f7a9c89da593346766b47c79

File: account_invoice_transmit_method/models/partner.py (hunk 5 above)
Comment (phuctranfxvn) [reply account_invoice_transmit_method/models/partner.py]: fixed in https://github.com/OCA/account-invoicing/pull/861/commits/da1b589510f39991f7a9c89da593346766b47c79

**Review by jcoux — APPROVED** [review]
Technically good for me :+1: 
(Maybe you can squash a little more commits from previous version to have more luck to have a merged PR)

**Comment by phuctranfxvn** [comment]
> Technically good for me 👍
> (Maybe you can squash a little more commits from previous version to have more luck to have a merged PR)

Done, thanks

File: account_invoice_transmit_method/models/partner.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    customer_invoice_transmit_method_id = fields.Many2one(
+        "transmit.method",
+        string="Customer Invoice Transmission Method",
+        company_dependant=True,
```
Comment (simahawk) [inline account_invoice_transmit_method/models/partner.py:14]: this is wrong (in v12 and v13 as well)
```suggestion
        company_dependent=True,
```
please check https://github.com/OCA/account-invoicing/pull/822 and maybe help that one get merged :pray:

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by alexis-via** [comment]
This is a duplicate of my migration PR https://github.com/OCA/account-invoicing/pull/823
The error mentionned by @simahawk is fixed in my PR, so I propose to cancel this one.

**Comment by simahawk** [comment]
> This is a duplicate of my migration PR #823
> The error mentionned by @simahawk is fixed in my PR, so I propose to cancel this one.

thanks for the pointer @alexis-via  ! :+1:

**Comment by simahawk** [comment]
@phuctranerp @jcoux  could you please review #823? Thanks!

**Pull request closed without merging**